notification/slack/channel.go

```diff
@@ -149,10 +162,15 @@ func (s *ChannelSender) loadChannel(ctx context.Context, channelID string) (*Cha
 		return nil, wrapError(resData.Error, "lookup Slack channel")
 	}
 
-	return &Channel{
-		ID:   resData.Channel.ID,
-		Name: "#" + resData.Channel.Name,
-	}, nil
+	if teamIDStr, ok := teamID.(*string); !ok {
+		return nil, wrapError(resData.Error, "lookup Slack channel")
+	} else {
```

Comment:
Don't use `else` if we're returning, also this check should happen up where we're actually doing the lookup